### PR TITLE
Bigger upstream merge batch

### PR DESCRIPTION
**Pull request by kmoppel-cognite**
Seems like a lot of stuff but should be no breaking changes. Main things added:

* Some new "approximate size" metrics that skip the real file-system backed functions
* Use approximate table / DB sizes automatically for bigger (1TB+) managed Azure Single Server instances due to FS lag
* New --emergency-pause-triggerfile flag to make it possible to pause all metrics gathering outside of normal automation / deployment
* New --no-helper-functions flag to ignore all metrics relying on 'get_smth()' functions and try the SU version instead. Tuned for managed instances.
* New --try-create-listed-exts-if-missing flag to to try to auto-create popular extensions outside of the helpers system that is not applicable on public clouds
* Better Prometheus mode cache and connection pool handling

### CODE REVIEW

**Comment by pashagolub** [comment]
What is the purpose of `/* pgwatch2_generated */` comments in SQL files? To determine `pgwatch2` queries among others during monitoring?

File: pgwatch2/metrics/settings/10/metric.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  current_setting('server_version_num')::int8 as server_version_num,
+  (regexp_matches(regexp_replace(current_setting('server_version'), '(beta|devel).*', '', 'g'), E'\\d+'))[1]::float8 as major_version,
+  current_setting('block_size')::int as block_size,
+  current_setting('max_connections')::int as max_connections,
+  current_setting('hot_standby') as hot_standby,
+  (select setting from qs where name = 'hot_standby_feedback') as hot_standby_feedback,
+  current_setting('fsync') as fsync,
+  current_setting('full_page_writes') as full_page_writes,
+  current_setting('synchronous_commit') as synchronous_commit,
+  (select setting from qs where name = 'wal_compression') as wal_compression,
```
Comment (pashagolub) [inline pgwatch2/metrics/settings/10/metric.sql:16]: Why cannot we use `select current_setting('wal_compression')` here?

File: pgwatch2/metrics/stat_statements/9.2/metric.sql [hunk 2]
```diff
@@ -70,7 +70,7 @@ FROM (
             total_time DESC
         LIMIT 100) a
 UNION
-SELECT
+select /* pgwatch2_generated */
```
Comment (pashagolub) [inline pgwatch2/metrics/stat_statements/9.2/metric.sql:73]: Not a bug bat rather code style. Uppercase `SELECT` was replaced with lower cases but all other keywords are still uppercased. I'm for uppercase keywords personally.

**Review by pashagolub — COMMENTED** [review]
Kudos for a huge amount of work! Are these changes were tested somewhere in a field?

File: pgwatch2/metrics/settings/10/metric.sql (hunk 1 above)
Comment (kmoppel-cognite) [reply pgwatch2/metrics/settings/10/metric.sql:16]: Looks ugly I know...but actually has some reasoning behind it...this is a newish parameter and with the current_setting() function you'll get a hard ERROR if the setting is not recognized...

File: pgwatch2/metrics/stat_statements/9.2/metric.sql (hunk 2 above)
Comment (kmoppel-cognite) [reply pgwatch2/metrics/stat_statements/9.2/metric.sql:73]: `/* pgwatch2_generated */  is indeed just to not exclude monitoring-generated stuff from dashboards etc. But agreed, should adhere to a unified style...uppercase probably indeed.

**Comment by kmoppel-cognite** [comment]
@pashagolub Yes, it's running at Cognite in production for 100+ DBs with no problems...at least so far :)

File: pgwatch2/metrics/stat_statements/9.2/metric.sql (hunk 2 above)
Comment (kmoppel-cognite) [reply pgwatch2/metrics/stat_statements/9.2/metric.sql:73]: Btw Pavlo, do you maybe have an idea how to automatically rewrite all those SQL queries - just good old `grep -i` or there's something better?

File: pgwatch2/metrics/settings/10/metric.sql (hunk 1 above)
Comment (pashagolub) [reply pgwatch2/metrics/settings/10/metric.sql:16]: Aha, so for old PG versions we're using `SELECT` from CTE and for fresh `current_setting()` directly?

**Pull request merged**